mkosi/backend.py

```diff
@@ -619,6 +621,48 @@ def get_partition(self, ident: PartitionIdentifier) -> Optional[Partition]:
             return None
         return self.partition_table.partitions.get(ident)
 
+    def _output_split_path(self, path_pattern: Optional[Path], partition_identifier: PartitionIdentifier) -> Path:
+        assert path_pattern is not None
+        partition = self.get_partition(partition_identifier)
+        assert self.partition_table is not None and partition is not None
+        part_size = partition.n_sectors * self.partition_table.sector_size
+        read_only = "1" if partition.read_only else "0"
+        wildcards = dict(s=str(part_size), r=read_only)
+        if self.config.image_version is not None:
+            wildcards["v"] = self.config.image_version
+        if partition.part_uuid is not None:
+            wildcards["u"] = str(partition.part_uuid)
+        name = NameTemplate(path_pattern.name).substitute(**wildcards)
```

Comment:
I choose `substitute` because it errors if an unknown wildcard is encountered. I find this preferable to silently fixing the string and generating artifacts with unexpected names. I've now added a nice error message for this case.